components/availability/src/Availability.svelte

```diff
@@ -334,27 +347,58 @@
     return epochs;
   }
 
-  $: days = scaleTime()
-    .domain([startDay, endDay])
-    .ticks(timeDay)
-    .filter((timestamp: Date) => {
-      if (show_weekends) {
-        return true;
-      } else {
-        return (
-          timestamp.toString() !== timeSaturday(timestamp).toString() &&
-          timestamp.toString() !== timeSunday(timestamp).toString()
-        );
+  function generateDayRange({
+    startDay,
+    endDay,
+    reverse = false,
+  }: {
+    startDay: Date;
+    endDay: Date;
+    reverse?: boolean;
+  }) {
+    let range = scaleTime()
+      .domain([startDay, endDay])
+      .ticks(timeDay)
+      .filter((timestamp: Date) => {
+        if (show_weekends) {
+          return true;
+        } else {
+          return (
+            timestamp.toString() !== timeSaturday(timestamp).toString() &&
+            timestamp.toString() !== timeSunday(timestamp).toString()
+          );
+        }
+      });
+    if (!show_weekends) {
+      let weekdayDates = range.filter(
+        (date) => date.getDay() !== 6 || date.getDay() !== 0,
+      );
+      if (weekdayDates.length < dates_to_show) {
+        if (reverse) {
```

Comment:
`reverse` flag is used when selecting previous dates / clicking the previous arrow. This lets the recursion from generateDayRange() attempt to move to **earlier** dates, when otherwise it would look further into the future.